### PR TITLE
Dns switch to endpointslices

### PR DESCRIPTION
**Pull request by aojea**
/kind cleanup

### CODE REVIEW

**Comment by aojea** [comment]
/assign @wojtek-t 
/cc @dpasiukevich 

It took me more than expected but we finally get it working

File: verify/verify-lint.sh [hunk 1]
```diff
@@ -37,6 +37,8 @@ MODULE_BASED=$(find . -type d -name vendor -prune \
 set +e
 status=0
 
+golangci-lint version
```
Comment (dpasiukevich) [inline verify/verify-lint.sh:40]: Is it needed?

**Review by dpasiukevich — COMMENTED** [review]
Thanks for the PR! I'm curious - what is the root cause of the lint failure?

File: verify/verify-lint.sh (hunk 1 above)
Comment (aojea) [reply verify/verify-lint.sh:40]: is it useful for debugging

**Comment by aojea** [comment]
> Thanks for the PR! I'm curious - what is the root cause of the lint failure?

the golangci-lint version was old

https://github.com/kubernetes/test-infra/pull/29690

**Review by dlapcevic — APPROVED** [review]
lgtm

**Comment by dlapcevic** [comment]
We need approval from OWNERS (https://github.com/kubernetes/perf-tests/blob/master/OWNERS).
@wojtek-t could you please approve?

**Comment by aojea** [comment]
/assign @marseel @mborsz 

they may help us here too

**Comment by mborsz** [comment]
/lgtm
/approve

BTW: Maybe we can remove vendor completely? I don't think it's needed nowadays. Sample PR for other components removing vendor: https://github.com/kubernetes/perf-tests/pull/1934

**Comment by aojea** [comment]
> BTW: Maybe we can remove vendor completely? I don't think it's needed nowadays. Sample PR for other components removing vendor: #1934

I was carrying on with current state, I was tempted to do it honestly, does not seem necessary

**Comment by dlapcevic** [comment]
I just built and released the new image `1.4.0` of dnsperfgo and it's 56 MB in size. It grew 10 MB from the previous version `1.3.0`.
That's quite big for such a small app. Is it due to vendor libraries?

**Comment by dpasiukevich** [comment]
> Is it due to vendor libraries?

I expect it to be the case. The prev kubernetes versions were kinda old
```
k8s.io/apimachinery v0.20.2
k8s.io/client-go v0.20.2
```

And now it's 0.27.2.

**Comment by aojea** [comment]
```
cd dns/dnsperfgo
go build
ls -altrh 
-rwxr-x---  1 aojea primarygroup  45M Jun  7 09:57 dnsperfgo
go install github.com/nikolaydubina/go-binsize-treemap@latest
go tool nm -size dnsperfgo | go-binsize-treemap > binsize.svg

```
binsize

**Comment by aojea** [comment]
I still don't understand well the diff between the reported size and the actual size

**Pull request merged**